### PR TITLE
Openssh contrib

### PR DESCRIPTION
**Pull request by SoapGentoo**

### CODE REVIEW

File: net-misc/openssh/openssh-9.3_p1-r1.ebuild [hunk 1]
```diff
@@ -71,14 +38,12 @@ REQUIRED_USE+="test? ( !xmss )"
 
 # Blocker on older gcc-config for bug #872416
```
Comment (mattst88) [inline net-misc/openssh/openssh-9.3_p1-r1.ebuild:39]: Drop this line?

File: net-misc/openssh-contrib/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		across the Internet unencrypted, but it is. OpenSSH encrypts all traffic (including passwords)
+		to effectively eliminate eavesdropping, connection hijacking, and other network-level attacks.
+		Additionally, OpenSSH provides a myriad of secure tunneling capabilities, as well as a variety
+		of authentication methods.
+
+		The OpenSSH suite includes the ssh program which replaces rlogin and telnet, scp which
+		replaces rcp, and sftp which replaces ftp. Also included is sshd which is the server side of
+		the package, and the other basic utilities like ssh-add, ssh-agent, ssh-keysign, ssh-keyscan,
+		ssh-keygen and sftp-server. OpenSSH supports SSH protocol versions 1.3, 1.5, and 2.0.
+	</longdescription>
```
Comment (mgorny) [inline net-misc/openssh-contrib/metadata.xml:25]: Perhaps mention what's the "-contrib" version about here.

File: net-misc/openssh/openssh-9.3_p1-r1.ebuild [hunk 3]
```diff
@@ -128,19 +93,31 @@ PATCHES=(
...
+	local i enabled_eol_flags disabled_eol_flags
+	for i in hpn sctp X509; do
+		if has_version "net-misc/openssh[${i}]"; then
+			enabled_eol_flags+="${i},"
+			disabled_eol_flags+="-${i},"
+		fi
+	done
+
+	if [[ -n ${enabled_eol_flags} && ${OPENSSH_EOL_USE_FLAGS_I_KNOW_WHAT_I_AM_DOING} != yes ]]; then
+		ewarn "net-misc/openssh does not support USE='${enabled_eol_flags%,}' anymore. The"
```
Comment (mgorny) [inline net-misc/openssh/openssh-9.3_p1-r1.ebuild:105]: You shouldn't put `the` at the end of the line xP.

File: app-admin/lnav/lnav-0.11.1-r2.ebuild [hunk 4]
```diff
@@ -28,7 +28,7 @@ RDEPEND="
 # The tests use ssh-keygen and use dsa and rsa keys (which is why ssl is required)
 DEPEND="${RDEPEND}
 	test? (
-		net-misc/openssh[ssl]
+		virtual/openssh[ssl(+)]
```
Comment (mgorny) [inline app-admin/lnav/lnav-0.11.1-r2.ebuild:31]: Is there a case for virtual/openssl not having ssl flag?

File: virtual/openssh/openssh-9.3_p1.ebuild [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+DESCRIPTION="Virtual for net-misc/openssh and variants"
+
+SLOT="0"
+KEYWORDS="~alpha amd64 arm arm64 hppa ~ia64 ~loong ~m68k ~mips ppc ppc64 ~riscv ~s390 sparc x86 ~x64-cygwin ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+IUSE="hpn sctp ssl X509"
+
+RDEPEND="
+	|| (
+		~net-misc/openssh-${PV}[hpn(-)?,sctp(-)?,ssl(-)?,X509(-)?]
+		~net-misc/openssh-contrib-${PV}[hpn(-)?,sctp(-)?,ssl(-)?,X509(-)?]
```
Comment (mgorny) [inline virtual/openssh/openssh-9.3_p1.ebuild:15]: I think you actually need `=` for these flags. There's one revdep that does `-something` dep on virtual, so this dep wouldn't enforce it on the underlying implementation.

File: net-misc/openssh-contrib/openssh-contrib-9.3_p1.ebuild [hunk 6]
```diff
@@ -210,13 +210,13 @@ src_prepare() {
...
 		mkdir "${hpn_patchdir}" || die
 		cp $(printf -- "${DISTDIR}/%s\n" "${HPN_PATCHES[@]}") "${hpn_patchdir}" || die
 		pushd "${hpn_patchdir}" &>/dev/null || die
 		eapply "${WORKDIR}/${HPN_GLUE_PATCH}"
 		use X509 && eapply "${WORKDIR}/${X509_HPN_GLUE_PATCH}"
-		use sctp && eapply "${FILESDIR}"/${PN}-8.5_p1-hpn-${HPN_VER}-sctp-glue.patch
+		use sctp && eapply "${FILESDIR}"/openssh-8.5_p1-hpn-${HPN_VER}-sctp-glue.patch
 		popd &>/dev/null || die
 
 		eapply "${hpn_patchdir}"
```
Comment (thesamesam) [inline net-misc/openssh-contrib/openssh-contrib-9.3_p1.ebuild:222]: Below, for HPN_DISABLE_MTAES, please link to https://bugs.gentoo.org/354113#c6 so finally 12 years after someone pointed it out, the comment for the test case makes it into the ebuild

File: net-misc/openssh/openssh-9.3_p1-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,370 @@
...
+		-i regress/Makefile || die
+
+	tc-export PKG_CONFIG
+	local sed_args=(
+		-e "s:-lcrypto:$(${PKG_CONFIG} --libs openssl):"
+		# Disable fortify flags ... our gcc does this for us
+		-e 's:-D_FORTIFY_SOURCE=2::'
+	)
+
+	# The -ftrapv flag ICEs on hppa #505182
```
Comment (thesamesam) [inline net-misc/openssh/openssh-9.3_p1-r1.ebuild:155]: drop this from both, I've tested it and it's fine without

**Review by mgorny — APPROVED** [review]
LGTM

**Comment by floppym** [comment]
It took me a few minutes to figure out what this PR is meant to accomplish. Maybe update the commit messages for openssh and openssh-compat to better explain what is happening.

**Review by floppym — APPROVED** [review]

**Review by thesamesam — APPROVED** [review]

**Review by mattst88 — APPROVED** [review]

**Pull request closed without merging**